### PR TITLE
PLF-8392 Kill Shindig

### PR DESCRIPTION
**Pull request by trongtt**

### CODE REVIEW

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/AbstractGadgetToPortletPlugin.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+                        PageContext page = iterator.next();
+
+                        LOG.info("Remove gadgets in page: " + page.getKey().format());
+                        this.migratePage(page);
+                        LOG.info("Migrate done");
+                    }
+                    offset += limit;
+                }
+            } catch (Exception e) {
+                LOG.error("Error during migrate gadgets: " + e.getMessage(), e);
```
Comment (thomasdelhomenie) [inline component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/AbstractGadgetToPortletPlugin.java]:
```suggestion
                LOG.error("Error during gadgets migration for " + siteType.getName()  + " sites: " + e.getMessage(), e);
```

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/AbstractGadgetToPortletPlugin.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            RequestLifeCycle.begin(PortalContainer.getInstance());
+            try {
+
+                result = this.pageService.findPages(offset, limit, siteType, null, null, null);
+                if (result != null && result.getSize() > 0) {
+                    Iterator<PageContext> iterator = result.iterator();
+                    while (iterator.hasNext()) {
+                        PageContext page = iterator.next();
+
+                        LOG.info("Remove gadgets in page: " + page.getKey().format());
```
Comment (thomasdelhomenie) [inline component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/AbstractGadgetToPortletPlugin.java]:
```suggestion
                        LOG.info("Remove/Replace gadgets in page: " + page.getKey().format());
```

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/AbstractGadgetToPortletPlugin.java [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+                        this.migratePage(page);
+                        LOG.info("Migrate done");
```
Comment (thomasdelhomenie) [inline component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/AbstractGadgetToPortletPlugin.java]:
```suggestion
                        LOG.info("Done for gadgets removal/replacement in page: " +  page.getKey().format());
```

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeGadgetToPortletPlugin.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    public UpgradeGadgetToPortletPlugin(InitParams initParams) {
+        super(initParams);
+    }
+
+    @Override
+    public void processUpgrade(String oldVersion, String newVersion) {
+
+        LOG.info("Remove the Monitoring and Management pages");
+        this.removeNavs();
```
Comment (thomasdelhomenie) [inline component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeGadgetToPortletPlugin.java]: Shouldn't we add a global catch in the method to add a log explaining that the upgrade did not finish correctly ?

Comment (thomasdelhomenie) [inline component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeGadgetToPortletPlugin.java]: We should add a log after each step to say that the step has finished successfully.

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeGadgetToPortletPlugin.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        LOG.info("Migrate Gadgets on group pages");
+        this.migratePages(SiteType.GROUP);
+
+        LOG.info("Migrate Gadgets on portal site layout");
+        this.migratePortals(SiteType.PORTAL);
+
+        LOG.info("Migrate Gadgets on group sites layout");
+        this.migratePortals(SiteType.GROUP);
+
+        LOG.info("Migrate gadgets done");
```
Comment (thomasdelhomenie) [inline component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeGadgetToPortletPlugin.java]:
```suggestion
        LOG.info("Gadgets migration done");
```

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeGadgetToPortletPlugin.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+                                        InitParams initParams) {
+        super(settingService, modelDataStorage, pageService, navigationService, initParams);
+    }
+
+    public UpgradeGadgetToPortletPlugin(InitParams initParams) {
+        super(initParams);
+    }
+
+    @Override
+    public void processUpgrade(String oldVersion, String newVersion) {
```
Comment (thomasdelhomenie) [inline component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeGadgetToPortletPlugin.java]: Did you check that if the migration fails at some point, it starts again at the next startup ?

Comment (thomasdelhomenie) [inline component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeGadgetToPortletPlugin.java]: We should add a log at the beginning and at the end of each upgrade plugin, something like "== Start removing/replacing gadgets in pages"

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeRemoveGadgetPlugin.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+import org.exoplatform.container.xml.InitParams;
+import org.exoplatform.portal.config.model.*;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+
+import javax.jcr.Node;
+import javax.jcr.NodeIterator;
+import java.util.List;
+
+public class UpgradeRemoveGadgetPlugin extends UpgradeProductPlugin {
```
Comment (thomasdelhomenie) [inline component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeRemoveGadgetPlugin.java]: Indentation is not correct (to check everywhere).

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeRemoveGadgetPlugin.java [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+            gadgets.remove();
+            session.save();
+
+            LOG.info("Remove node exo:gadget-groovy");
+            gadgets = rootNode.getNode("exo:gadget-groovy");
+            gadgets.remove();
+            session.save();
+
+        } catch (Exception ex) {
+            LOG.error("Exception while remove gadgets", ex);
```
Comment (thomasdelhomenie) [inline component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeRemoveGadgetPlugin.java]:
```suggestion
            LOG.error("Exception while removing gadgets definitions", ex);
```

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/AbstractGadgetToPortletPlugin.java [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+                }
+                LOG.info("Migrate gadget done");
+
+                return null;
+            } else {
+                if ("dashboard/GadgetPortlet".equals(contentId)
+                        || "dashboard/TabbedDashboardPortlet".equals(contentId)
+                        || "dashboard/DashboardPortlet".equals(contentId)
+                ) {
+                    LOG.info("Remove portlet " + contentId + " from page");
```
Comment (thomasdelhomenie) [inline component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/AbstractGadgetToPortletPlugin.java]: This results in this kind of log:
```
2019-07-30 05:52:28,431 | INFO  | Remove portlet dashboard/TabbedDashboardPortlet from page [o.e.p.u.p.UpgradeGadgetToPortletPlugin<Catalina-startStop-1>] 
2019-07-30 05:52:28,840 | INFO  | Remove portlet dashboard/TabbedDashboardPortlet from page [o.e.p.u.p.UpgradeGadgetToPortletPlugin<Catalina-startStop-1>] 
2019-07-30 05:52:29,283 | INFO  | Remove portlet dashboard/TabbedDashboardPortlet from page [o.e.p.u.p.UpgradeGadgetToPortletPlugin<Catalina-startStop-1>] 
2019-07-30 05:52:29,709 | INFO  | Remove portlet dashboard/TabbedDashboardPortlet from page [o.e.p.u.p.UpgradeGadgetToPortletPlugin<Catalina-startStop-1>] 
2019-07-30 05:52:30,109 | INFO  | Remove portlet dashboard/TabbedDashboardPortlet from page [o.e.p.u.p.UpgradeGadgetToPortletPlugin<Catalina-startStop-1>] 
2019-07-30 05:52:30,534 | INFO  | Remove portlet dashboard/TabbedDashboardPortlet from page [o.e.p.u.p.UpgradeGadgetToPortletPlugin<Catalina-startStop-1>] 
2019-07-30 05:52:30,957 | INFO  | Remove portlet dashboard/TabbedDashboardPortlet from page [o.e.p.u.p.UpgradeGadgetToPortletPlugin<Catalina-startStop-1>] 
```
We should add on which page it is done, in order to have a relevant log.

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/AbstractGadgetToPortletPlugin.java [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+                } else if ("rssAggregator".equals(contentId)) {
+                    newContentId = "portlets/RSSReader";
+                } else if ("LoginHistory".equals(contentId)) {
+                    newContentId = "portlets/LoginHistory";
+                } else if ("FeaturedPoll".equals(contentId)) {
+                    newContentId = "portlets/FeaturedPoll";
+                }
+
+                if (newContentId != null) {
+                    LOG.info("== Replace by new portlet " + newContentId);
```
Comment (thomasdelhomenie) [inline component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/AbstractGadgetToPortletPlugin.java]: The `==` prefix in the log should be used for migration start and end logs IMO, in order to identify easily the start and the stop of the migration. For others logs, it is not necessary.

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/AbstractGadgetToPortletPlugin.java [hunk 11]
```diff
@@ -0,0 +1,236 @@
...
+    }
+
+    protected ComponentData migrateComponent(ComponentData component) throws Exception {
+        if (component instanceof ApplicationData) {
+            ApplicationData<?> app = (ApplicationData<?>)component;
+            ApplicationState state = app.getState();
+            String contentId = this.modelDataStorage.getId(state);
+
+            if (app.getType() == ApplicationType.GADGET) {
+                LOG.info("Migrate for gadget: " + contentId);
```
Comment (thomasdelhomenie) [inline component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/AbstractGadgetToPortletPlugin.java]: We should add the name of the page IMO.
```suggestion
                LOG.info("Gadget " + contentId + " found in page " + page);
```

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/AbstractGadgetToPortletPlugin.java [hunk 12]
```diff
@@ -0,0 +1,236 @@
...
+                    a.setShowInfoBar(app.isShowInfoBar());
+                    a.setTheme(app.getTheme());
+                    a.setTitle(app.getTitle());
+                    a.setDescription(app.getDescription());
+                    a.setWidth(app.getWidth());
+                    a.setHeight(app.getHeight());
+
+                    return (ApplicationData)a.build();
+                }
+                LOG.info("Migrate gadget done");
```
Comment (thomasdelhomenie) [inline component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/AbstractGadgetToPortletPlugin.java]: If we are here, it means that the gadget is removed, right ? If so, the log should be:
```suggestion
                LOG.info("Gagdet " + contentId + " removed from page " + page);
```

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeUserPortalPlugin.java [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    @Override
+    public void processUpgrade(String oldVersion, String newVersion) {
+
+//        LOG.info("Migrate Gadgets on user pages");
+//        this.migratePages(SiteType.USER);
+
+        LOG.info("Migrate Gadgets on user sites layout");
+        this.migratePortals(SiteType.USER);
```
[the remark this answers is not in the document]
Comment (trongtt) [reply component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeUserPortalPlugin.java]: currently we didn't remove user site just remove the pages and navigations only. Therefore we still need migrate for user site.
Do you think we should remove user site too?

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeGadgetToPortletPlugin.java [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+import org.exoplatform.container.xml.InitParams;
+import org.exoplatform.portal.mop.SiteKey;
+import org.exoplatform.portal.mop.SiteType;
+import org.exoplatform.portal.mop.navigation.*;
+import org.exoplatform.portal.mop.page.PageService;
+import org.exoplatform.portal.pom.data.*;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+
+public class UpgradeGadgetToPortletPlugin extends AbstractGadgetToPortletPlugin {
```
[the remark this answers is not in the document]
Comment (trongtt) [reply component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeGadgetToPortletPlugin.java]: We don't need these plugins run in order. I just separate into 3 plugins to make the code easier to read/maintain

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeUserPortalPlugin.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (thomasdelhomenie) [reply component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeUserPortalPlugin.java]: ok. User sites were only used for dashboard app, no ?

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeGadgetToPortletPlugin.java (hunk 14 above)
[the remark this answers is not in the document]
Comment (thomasdelhomenie) [reply component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeGadgetToPortletPlugin.java]: ok :+1:

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeUserPortalPlugin.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (trongtt) [reply component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeUserPortalPlugin.java]: ahh, you are right. Seem there is no way to access user site after we remove the dashboard.
So we should remove user site too.

[the remark this answers is not in the document]
Comment (thomasdelhomenie) [reply component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeUserPortalPlugin.java]: I am not sure actually. These sites are created on user creation ? If so, we will not be able to recreate them if necessary, no ?

File: portlets/src/main/webapp/login-history/filters/prettyDate.js [hunk 15]
```diff
@@ -0,0 +1,64 @@
+export default function prettyTimeDiff(aTime, shortFormat, smallest, bTime) {
```
[the remark this answers is not in the document]
Comment (trongtt) [reply portlets/src/main/webapp/login-history/filters/prettyDate.js]: No, that's javascript function I move it from the gadget to use.

[the remark this answers is not in the document]
Comment (thomasdelhomenie) [reply portlets/src/main/webapp/login-history/filters/prettyDate.js]: ok :+1:

File: portlets/src/main/java/org/exoplatform/platform/portlet/poll/FeaturedPollController.java [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+import juzu.template.Template;
+
+import javax.inject.Inject;
+import javax.portlet.ReadOnlyException;
+import javax.portlet.ValidatorException;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FeaturedPollController {
```
[the remark this answers is not in the document]
Comment (trongtt) [reply portlets/src/main/java/org/exoplatform/platform/portlet/poll/FeaturedPollController.java]: I need action method to allow save portlet preferences, so I use Juzu to take easy. Isn't it good? We still should use GenericPortlet?

[the remark this answers is not in the document]
Comment (thomasdelhomenie) [reply portlets/src/main/java/org/exoplatform/platform/portlet/poll/FeaturedPollController.java]: Yes we should use GenericPortlet instead. We want to get rid of juzu, so I prefer to not add new juzu portlet.

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeUserPortalPlugin.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (trongtt) [reply component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeUserPortalPlugin.java]: Right, I have checked again and user's portal only be created lazy when user access to dashboard page via UserPortalLifecycle but it's removed.
So no way to recreat it now I think.

File: portlets/src/main/java/org/exoplatform/platform/portlet/poll/FeaturedPollController.java (hunk 16 above)
[the remark this answers is not in the document]
Comment (trongtt) [reply portlets/src/main/java/org/exoplatform/platform/portlet/poll/FeaturedPollController.java]: Ok, so I will rework on this one.

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeUserPortalPlugin.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (thomasdelhomenie) [reply component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeUserPortalPlugin.java]: ok. Just to be sure, don't we have others features which rely on user sites ?

File: portlets/src/main/java/org/exoplatform/platform/portlet/poll/FeaturedPollController.java (hunk 16 above)
[the remark this answers is not in the document]
Comment (thomasdelhomenie) [reply portlets/src/main/java/org/exoplatform/platform/portlet/poll/FeaturedPollController.java]: :+1:

File: exo-gadget-pack/gadget-pack-services/src/main/java/org/exoplatform/platform/gadget/services/LoginHistory/LoginHistoryUpgradePlugin.java [hunk 17]
```diff
@@ -1,481 +0,0 @@
-package org.exoplatform.platform.gadget.services.LoginHistory;
```
[the remark this answers is not in the document]
Comment (trongtt) [reply exo-gadget-pack/gadget-pack-services/src/main/java/org/exoplatform/platform/gadget/services/LoginHistory/LoginHistoryUpgradePlugin.java]: I bring it back and move to "component/common" module.

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeUserPortalPlugin.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (trongtt) [reply component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/UpgradeUserPortalPlugin.java]: Yes, I'm quite sure that there is no other features rely on user sites.

File: portlets/src/main/webapp/bookmark/main.js [hunk 18]
```diff
@@ -0,0 +1,20 @@
+import './components/initComponents';
+import {exoConstants} from '../js/eXoConstants';
+
+// getting language of the PLF
+const lang = typeof eXo !== 'undefined' ? eXo.env.portal.language : 'en';
+
```
[the remark this answers is not in the document]
Comment (trongtt) [reply portlets/src/main/webapp/bookmark/main.js]: It's done.

**Pull request merged**